vkcs/compute/resource_instance.go

```diff
@@ -405,6 +405,40 @@ func ResourceComputeInstance() *schema.Resource {
 	}
 }
 
+func resourceComputeInstanceCreateWithWarnings(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
I see no reason to write it in a different function, if you want to separate the code below from the `resourceComputeInstanceCreate`, then you should write function that returns diagnostics with warnings and add them to diagnostics in the main function.

Reply:
I move new checks to `checkBlockDeviceConfig`, which calls in `resourceComputeInstanceCreate`